### PR TITLE
Hover must work not only if onTap enabled

### PR DESCRIPTION
**Pull request by Oliinyk-Volodymyr**

### CODE REVIEW

File: packages/flutter/lib/src/material/ink_well.dart [hunk 1]
```diff
@@ -38,29 +38,32 @@ abstract class InteractiveInkFeature extends InkFeature {
...
+  void cancel() {}
 
   /// The ink's color.
   Color get color => _color;
   Color _color;
+
   set color(Color value) {
-    if (value == _color)
-      return;
+    if (value == _color) return;
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/material/ink_well.dart]: This change is against the style guide (the unchanged one is actually ok) https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#separate-the-if-expression-from-its-statement

There are more cases later.

File: packages/flutter/lib/src/material/ink_well.dart [hunk 2]
```diff
@@ -449,7 +452,8 @@ class InkResponse extends StatefulWidget {
   }
 
   @override
-  _InkResponseState<InkResponse> createState() => _InkResponseState<InkResponse>();
+  _InkResponseState<InkResponse> createState() =>
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/material/ink_well.dart]: Actually please keep these statements in a single line if that is clearer. Flutter prefers an 80-column line but doesn't enforce so, if breaking the line makes it less clear. There are many more cases later.

**Review by dkwingsmt — CHANGES_REQUESTED** [review]
Why do we need this change? To my understanding, `InkWell` uses `onTap` to determine if it's enabled, and a disabled inkwell should not react to hover events or show hightlight. 
Additionally, please add tests to avoid regression.
Also, please avoid using automatic formatting tool, especially using it massively on existing files. https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#in-defense-of-the-extra-work-that-hand-formatting-entails

File: packages/flutter/lib/src/material/ink_well.dart (hunk 1 above)
Comment (Denive) [reply packages/flutter/lib/src/material/ink_well.dart]: Inkwell have method onHover, i think him must work without onTap.

**Comment by Denive** [comment]
Inkwell have method onHover, i think him must work without onTap.

**Comment by dkwingsmt** [comment]
That's a good point. I'm wondering if the better way might be to make it "enabled" if either `onTap` or `onHover` is available. /cc @HansMuller

**Comment by gspencergoog** [comment]
No, a disabled control (which is what you have if `onTap` isn't defined) doesn't have any hover behavior: it's disabled, so shouldn't respond to hover or anything else. This comes from the states in the [Material spec for buttons](https://material.io/design/interaction/states.html#disabled). If you want to know if the mouse is over a disabled button, you would have to wrap it in a `MouseRegion`.

**Pull request closed without merging**